extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentKiller.java

```diff
@@ -99,13 +118,20 @@ public void killAll() throws IOException
     fs.delete(storageDirectory, true);
   }
 
-  private boolean safeNonRecursiveDelete(FileSystem fs, Path path)
+  private void deleteEmptyParents(FileSystem fs, Path path)
   {
+    if(path.equals(storageDirectory)) {
```

Comment:
this does not work if storageDirectory has changed since the time segment, being deleted, was created.